File: experiments/016-LanguageComparisonRepeated/cells/java-gradle_1-Empty_free/2019-2c-parcial2-masked/20260917-053838-37234/analysis/acceptance-project/src/main/java/surveycrawler/Silt.java
package surveycrawler;

import java.util.Random;

public class Silt extends GroundType {

    public static final int SLIDING_LIMIT = 10;

    private final Random random;

    // initialization

    public Silt(Random aRandom) {
        random = aRandom;
    }

    // moving

    @Override
    public void moveTowards(SurveyCrawler aCrawler, Point aDirection) {
        aCrawler.slideBy(aDirection.times(slidingDistance()));
    }

    public int slidingDistance() {
        return random.nextInt(SLIDING_LIMIT) + 1;
    }

    // facing

    @Override
    public void assertCanTurn(SurveyCrawler aCrawler) {
    }
}

File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2019-2c-parcial2-masked/20260921-204513-77784/output/project/src/surveycrawler/silt.py
import random

from surveycrawler.ground import Ground


class Silt(Ground):

    # instance creation

    @classmethod
    def sliding_randomly(cls):
        return cls(random.Random())

    @classmethod
    def sliding_with(cls, a_random):
        return cls(a_random)

    # initialization

    def __init__(self, a_random):
        self._random = a_random

    # sliding

    @classmethod
    def max_number_of_cells_to_slide(cls):
        return 10

    def number_of_cells_to_slide(self):
        return self._random.randrange(self.max_number_of_cells_to_slide()) + 1

    # moving

    def move_crawler_towards(self, a_crawler, a_direction):
        a_crawler.slide_towards(a_direction, self)

    # facing

    def assert_crawler_can_turn(self, a_crawler):
        pass
